modules/apps/translation/translation-translator-deepl/src/main/java/com/liferay/translation/translator/deepl/internal/translator/DeepLTranslator.java

```diff
@@ -0,0 +1,246 @@
+			_configurationProvider.getCompanyConfiguration(
+				DeepLTranslatorConfiguration.class,
+				translatorPacket.getCompanyId());
+
+		String targetLanguageCode = _getLanguageCode(
+			translatorPacket.getTargetLanguageId());
+
+		List<String> supportedLanguages = _getSupportedLanguages(
+			deepLTranslatorConfiguration);
+
+		if (!_verifyLanguage(supportedLanguages, targetLanguageCode)) {
+			_log.error(
+				StringBundler.concat(
+					"No target language available for ", targetLanguageCode,
+					". Supported languages are: ",
+					StringUtil.merge(
+						supportedLanguages, StringPool.COMMA_AND_SPACE)));
+
+			return translatorPacket;
+		}
+
+		String slc = translatorPacket.getSourceLanguageId();
+
+		String sourceLanguageCode = _getLanguageCode(slc);
+
+		Map<String, String> translatedFieldsMap = new HashMap<>();
+		Map<String, String> fieldsMap = translatorPacket.getFieldsMap();
+
+		for (Map.Entry<String, String> entry : fieldsMap.entrySet()) {
+			translatedFieldsMap.put(
+				entry.getKey(),
+				_translate(
+					deepLTranslatorConfiguration.url(),
+					deepLTranslatorConfiguration.authKey(), entry.getValue(),
+					sourceLanguageCode, targetLanguageCode));
+		}
+
+		return new TranslatorPacket() {
+
+			@Override
+			public long getCompanyId() {
+				return translatorPacket.getCompanyId();
+			}
+
+			@Override
+			public Map<String, String> getFieldsMap() {
+				return translatedFieldsMap;
+			}
+
+			@Override
+			public String getSourceLanguageId() {
+				return translatorPacket.getSourceLanguageId();
+			}
+
+			@Override
+			public String getTargetLanguageId() {
+				return translatorPacket.getTargetLanguageId();
+			}
+
+		};
+	}
+
+	@Activate
+	@Modified
+	protected void activate(Map<String, Object> properties) {
+		_deepLTranslatorConfiguration = ConfigurableUtil.createConfigurable(
+			DeepLTranslatorConfiguration.class, properties);
+	}
+
+	private String _getLanguageCode(String languageId) {
+		List<String> list = Arrays.asList(
+			StringUtil.split(languageId, CharPool.UNDERLINE));
+
+		String ret = list.get(0);
+
+		return StringUtil.toUpperCase(ret);
+	}
+
+	private List<String> _getSupportedLanguages(
+		DeepLTranslatorConfiguration deepLTranslatorConfiguration) {
+
+		List<SupportedLanguage> supportedLanguages = null;
+
+		try {
+			supportedLanguages = _deepLClient.verifySupportedLanguage(
+				deepLTranslatorConfiguration.authKey(), DeepLConstants.TARGET,
+				DeepLConstants.SUPPORTED_LANGUAGE_INQ_URL);
+		}
+		catch (IOException ioException) {
+			_log.error(
+				"Failed to call supported language list." +
+					System.lineSeparator() + ioException.getLocalizedMessage());
+
+			return new ArrayList<>();
+		}
+		catch (TranslatorException translatorException) {
+			throw new RuntimeException(translatorException);
+		}
+
+		List<String> languages = new ArrayList<>();
+
+		supportedLanguages.forEach(sl -> languages.add(sl.getLanguage()));
+
+		return languages;
```

Comment:
Can you increase the scope of the try/catch block so that all code representing the happy path is in the try section?

Also, rename `sl` so that it is fully spelled.